contracts/src/core/EigenDAServiceManager.sol

```diff
@@ -140,42 +123,53 @@ contract EigenDAServiceManager is Initializable, OwnableUpgradeable, EigenDAServ
 
-        emit BatchConfirmed(reducedBatchHeaderHash, batchIdMemory, fee);
+        emit BatchConfirmed(reducedBatchHeaderHash, batchIdMemory, fee, false);
 
         // increment the batchId
         batchId = batchIdMemory + 1;
     }
 
-    /**
-     * @notice Forwards a call to EigenLayer's DelegationManager contract to confirm operator registration with the AVS
-     * @param operator The address of the operator to register.
-     * @param operatorSignature The signature, salt, and expiry of the operator's signature.
-     */
-    function registerOperatorToAVS(
-        address operator,
-        ISignatureUtils.SignatureWithSaltAndExpiry memory operatorSignature
-    ) external {
-        delegationManager.registerOperatorToAVS(operator, operatorSignature);
-    }
+    /// @notice This function is used for submitting data availabilty certificates optimistically
+    function confirmBatchOptimistically(
+        BatchHeader calldata batchHeader
+    ) external onlyWhenNotPaused(PAUSED_CONFIRM_BATCH) onlyBatchConfirmer() {
+        // make sure the information needed to derive the non-signers and batch is in calldata to avoid emitting events
+        require(tx.origin == msg.sender, "EigenDAServiceManager.confirmBatch: header and nonsigner data must be in calldata");
+        // make sure the stakes against which the Batch is being confirmed are not stale
+        require(
+            batchHeader.referenceBlockNumber <= block.number, "EigenDAServiceManager.confirmBatch: specified referenceBlockNumber is in future"
+        );
 
-    /**
-     * @notice Forwards a call to EigenLayer's DelegationManager contract to confirm operator deregistration from the AVS
-     * @param operator The address of the operator to deregister.
-     */
-    function deregisterOperatorFromAVS(address operator) external {
-        delegationManager.deregisterOperatorFromAVS(operator);
-    }
+        require(
+            (batchHeader.referenceBlockNumber + BLOCK_STALE_MEASURE) >= uint32(block.number),
+            "EigenDAServiceManager.confirmBatch: specified referenceBlockNumber is too far in past"
+        );
 
-    /**
-     * @notice Sets the metadata URI for the AVS
-     * @param _metadataURI is the metadata URI for the AVS
-     */
-    function setMetadataURI(string memory _metadataURI) external onlyOwner() {
-        metadataURI = _metadataURI;
+        // calculate reducedBatchHeaderHash which nodes signed
+        bytes32 reducedBatchHeaderHash = batchHeader.hashBatchHeaderToReducedBatchHeader();
+
+        // store the metadata hash
+        uint96 fee = 0;
+        uint32 batchIdMemory = batchIdOptimistic;
+        bytes32 batchHeaderHash = batchHeader.hashBatchHeader();
+        batchIdToBatchMetadataHashOptimistic[batchIdMemory] = EigenDAHasher.hashBatchHashedMetadata(batchHeaderHash, fee, uint32(block.number));
+
+        emit BatchConfirmed(reducedBatchHeaderHash, batchIdMemory, fee, true);
```

Comment:
this might mess up our indexing to add another param, so another event might be preferrable.

@mooselumph wdyt

Reply:
I don't see why it should complicated it, but I also don't understand what the `confirmBatchOptimistically` function is for. Can someone explain?